src/compiler/checker.ts

```diff
@@ -15845,6 +15982,7 @@ namespace ts {
                 checkExportsOnMergedDeclarations(node);
                 const symbol = getSymbolOfNode(node);
                 checkTypeParameterListsIdentical(node, symbol);
+                updateReferencesForInterfaceHeritiageClauseTargets(node);

```

Comment:
Agreed that checkUnusedTypeParameters(node) has to be added here. But, I am adding it a little down after "forEach(node.members, checkSourceElement);" So that the body will be completely processed before checking the hasReference of the type parameter